File: Downloads/NewBalanceStoreBackend-sprint4-pr/NewBalanceStoreBackend-sprint4-pr/frontend/src/entities/Product/Product.js
export class Product {
    id;
    images;
    name;
    oldPrice;
    price;
    sizes;
    colors;
    gender;
    type;
    category;
    activity;
    description;
    isNew;
    inStock;

    constructor(data) {
        Object.assign(this, data);
    }

    getMainImage(color = null) {
        const imageColor = color ?? this.colors[0];
        return this.images[imageColor][0];
    }

    isAvailable() {
        return this.inStock !== false;
    }

    hasDiscount() {
        return this.oldPrice !== null;
    }
}

File: Downloads/NewBalanceStoreBackend-sprint4-pr/NewBalanceStoreBackend-sprint4-pr/backend/NewBalanceStore.Domain/Entities/Product.cs
using System;
using System.Collections.Generic;
using System.Text;

using NewBalanceStore.Domain.Enums;

namespace NewBalanceStore.Domain.Entities;

public class Product
{
    public string Id { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public ProductType Type { get; set; }
    public string Category { get; set; } = string.Empty;
    public Gender Gender { get; set; }
    public decimal Price { get; set; }
    public decimal? OldPrice { get; set; }
    public List<string> Images { get; set; } = new();
    public bool IsNew { get; set; }
    public List<ProductVariant> Variants { get; set; } = new();
}